doc/Manuals/ug-ss_x-road_6_security_server_user_guide.md

```diff
@@ -2143,18 +2146,83 @@ For example, these log messages are related to an API call with correlation ID `
 
 ### 19.4 Validation errors
 
-An error response from the REST API can include validation errors if an unsupported parameter was provided with the request. 
+An error response from the REST API can include validation errors if an unsupported parameter was provided with the request.
+When 
 
-In addition to the validation messages declared in Java Validation API, the following validation errors are possible:
+Example request and response of adding a new subsystem with illegal characters:
 ```
-Normalized
+POST https://ss1:4100/api/v1/clients
+
+Request body:
+{
+  "client": {
+    "member_class": "ORG",
+    "member_code": "0/1234",
+    "subsystem_code": "Subsystem%Code"
+  },
+  "ignore_warnings": false
+}
+
+Response body:
+{
+  "error": {
+    "code": "validation_failure",
+    "validation_errors": {
+      "clientAdd.client.memberCode": [
+        "NoForwardslashes"
+      ],
+      "clientAdd.client.subsystemCode": [
+        "NoPercents"
+      ]
+    }
+  },
+  "status": 400
+}
+```
+
+In addition to the validation messages declared in [Java Validation API](https://javaee.github.io/javaee-spec/javadocs/javax/validation/constraints/package-summary.html), the following validation errors are possible:
+```
+NoControlChars
 NoColons
 NoSemicolons
 NoForwardslashes
 NoBackslashes
 NoPercents
 ```
 
+### 19.5 Warning responses
+
+Error response from the Management API can include additional warnings that you can ignore if seen necessary. The warnings can be ignored by your decision, by executing the same operation with `ignore_warnings` boolean parameter set to `true`. *Always consider the warning before making the decision to ignore it.* 
+
+An example case:
+1. Client executes a REST request, without `ignore_warnings` parameter, to backend
+2. Backend notices warnings and responds with error message that contains the warnings. Nothing is updated at this point
+3. Client determines if warnings can be ignored. If the request was sent via the admin UI, a popup with `CONTINUE` button is shown
+4. If the warnings can be ignored, client resends the REST request, but with `ignore_warnings` parameter set to `true`. If the request was sent via the admin UI and a warning popup with the `CONTINUE` button is shown, clicking the button will ignore warnings and proceed to executing the operation
```

Comment:
Suggestion, would slight rewording make this easier to understand? 

> 4. If the warnings can be ignored, client resends the REST request, but with `ignore_warnings` parameter set to `true`. If the request was sent via the admin UI and a warning popup with the `CONTINUE` button is shown, clicking the button will ignore warnings and proceed to executing the operation


->

> 4. If the warnings can be ignored, client resends the REST request, but with `ignore_warnings` parameter set to `true`. If the original request was sent via the admin UI and a warning popup with the `CONTINUE` button was shown, clicking the button will cause this resending to happen.

Or, alternatively, maybe remove mentions of admin UI from steps 1-5 and explain how admin UI handles this via popup, after 1-5. At current form step 4 is maybe a bit confusing as it combines explaining "general client" and "concrete admin UI" cases, I feel.

Reply:
Separated the UI usage from the steps and moved it to the [end of the chapter](https://github.com/nordic-institute/X-Road/pull/720/files#diff-384166854dd99feefcf084d956dc2f1c23e6764e5802dc32ef8bf2e8557622f2R2396)